lib/launchers/v_sphere.rb

```diff
@@ -391,6 +391,33 @@ def object.conwrapper
       end
     end
 
+    # @return Array of VirtualMachine objects
+    private def vms(folder, machines)
```

Comment:
It would suggest this method to take only one argument `folder`. Then have in body
```
private def vms(folder: nil)
folder ||= datacenter.vmFolder
machines = []
...
        when "Folder"
          machines.concat vms(x)
...
return machines
```

Then you can alter method `#get_running_instances` to filter out only `poweredOn` instances and return. This will make for less boilerplate
```
    def get_running_instances
      return vms.select { |i| i.runtime.powerState == "poweredOn" }
    end
```

Reply:
@akostadinov done...

Reply:
hmm, forgot to push? I still see two args.